### PR TITLE
Updated to default boost 1_59, added Google Protocol buffers v3 and MySQL C Connector

### PR DESCRIPTION
**Pull request by dlinten**
I have updated the default boost version to 1_59.
I have added Google Protocol Buffers v3
I have added MySQL C Connector v6.1.6

### CODE REVIEW

File: cmake/projects/protobuf/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    protobuf
+    VERSION
+    "v3.0.0-hunter-p1"
+    URL
+    "https://github.com/dlinten/protobuf/archive/v3.0.0-hunter-p1.tar.gz"
```
Comment (ruslo) [inline cmake/projects/protobuf/hunter.cmake]: There is a discussion about `protobuf` already: https://github.com/ruslo/hunter/issues/97 and repository for patches: https://github.com/hunter-packages/protobuf please send your local patches there to branch `hunter` so we can keep this stuff central. I've send you invitation to the team, check your email.

File: cmake/projects/Boost/hunter.cmake [hunk 2]
```diff
@@ -21,6 +21,18 @@ hunter_add_version(
     PACKAGE_NAME
     Boost
     VERSION
+    "1.59"
+    URL
+    "http://sourceforge.net/projects/boost/files/boost/1.59.0/boost_1_59_0.tar.gz/download"
```
Comment (ruslo) [inline cmake/projects/Boost/hunter.cmake]: Official boost archives are too bloated. I'm using this repo to store unpacked version and remove docs and tests directories: https://github.com/hunter-packages/boost This makes it like 3 times smaller. Doing it now for 1.59, also there is 1.60 release.

File: cmake/projects/MySQLCConnector/hunter.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    MySQLCConnector
+    VERSION
+    "v6.1.6"
+    URL
+    "https://dev.mysql.com/get/Downloads/Connector-C/mysql-connector-c-6.1.6-src.tar.gz"
```
Comment (ruslo) [inline cmake/projects/MySQLCConnector/hunter.cmake]: Can you please create pull request will only this change, will merge it. Also please set destination branch to `develop`, see https://github.com/ruslo/hunter/wiki/dev.contribution

File: cmake/projects/protobuf/hunter.cmake (hunk 1 above)
Comment (ruslo) [inline cmake/projects/protobuf/hunter.cmake]: > please send your local patches there to branch hunter

Even if you have permissions to pull to branch directly it's better to send pull request for review reasons.

File: cmake/projects/Boost/hunter.cmake (hunk 2 above)
Comment (ruslo) [inline cmake/projects/Boost/hunter.cmake]: You can use these releases:
* https://github.com/hunter-packages/boost/releases/tag/v1.60.0
* https://github.com/hunter-packages/boost/releases/tag/v1.59.0

**Comment by ruslo** [comment]
@dlinten Hi, what is the status of this PR. I see some new patches not related to original PR.

**Comment by dlinten** [comment]
@ruslo No further updates have been made as of yet.  Please ignore PR.  I will redo and conform to review procedures.

**Comment by dlinten** [comment]
Performing redo in accordance with review procedures.

**Comment by ruslo** [comment]
just to summarize:
* mysql connector add intent: https://github.com/ruslo/hunter/pull/316/files
* protobuf: https://github.com/ruslo/hunter/issues/97. I see that already like 3 developers try to add it. Project use CMake as a build system and looks pretty trivial, not sure what is the problem here...
* I've made a "cut" boost releases for 1.59, 1.60 versions. All that need to be done is to calculate sha1 of archives

**Pull request closed without merging**